### DISCUSSION TITLE
3XX Field Mappings

### DISCUSSION THREAD

**Discussion by cspayne** [crystalyragui/MARC2RDA#456] [General]
Here's a place to discuss 3XX mappings as we update them and ask questions! 

Previous discussion in: 
[344 issue](https://github.com/uwlib-cams/MARC2RDA/issues/153#issuecomment-2208801874)
[2024-07-04 M2R meeting](https://github.com/uwlib-cams/MARC2RDA/issues/153#issuecomment-2208801874)
[$2 issue](https://github.com/uwlib-cams/MARC2RDA/issues/439)

**Comment by cspayne** [comment]
Penny:
> Hi all. For the 344 $2 mapping, does it need to mint a nomen and then use the element "has scheme of nomen"? For example, for $a, since the element "type of recording" does not have a range, how can I associate it with a nomen? And, will $0 and $1 be the identifier or URI for the value in $a? If they are URIs, can they be used directly as values and $2 be discarded? If $0 is a identifier, how can we map the source or schema?

Gordon: 

>

>  @pennylenger: We should not use a nomen as the value of an attribute element like field 344.
> 
> Instead, use skos:Concept as the type of the minted IRI, use skos:prefLabel for the attribute value, and use skos:inScheme for the source. This is consistent with the approach used for subject field concepts.
> 
> We have to assume, I think, that subfield $1 references the concept as a rwo, and subfield $0 is the usual muddle. This is still under general discussion ...
> 
> So I would treat subfield $1 as the object value of the appropriate RDA element and ignore the other subfields. If there is no subfield $1 but there is a string value and a subfield $2 source, I would mint a skos:Concept as the object value and add the prefLabel and inScheme statements.

**Comment by cspayne** [comment]
Penny:

> @GordonDunsire Thank you very much, Gordon. I have another question. For fields such as 336, 337, and 338, where $b is the code for the term in $a, if there is no $1 and $0 is not a URI, should $b and $a be mapped together by minting a skos
> as the object value, or separately? Or is there another mapping method for the code? And if rdf:datatype is used, is it more concise than minting a skos:Concept as the object value and add the prefLabel and inScheme statements?

Gordon:

> @pennylenger: Use skos:notation for the code. To clarify, for fields that use subfield $b for a code for a term from a controlled vocabulary, and:
> 
>     For instance data that has a value in subfield $1, use the appropriate RDA object property with the IRI in subfield $1 and do not transform the other subfields.
>     For instance data that has no value in subfield $1 but has a value in subfield $2 (source) that is not an RDA vocabulary, mint a skos:Concept and use the appropriate RDA object property with the minted IRI. Transform the subfields to skos properties of the minted IRI as follows:
>     -- subfield $a : skos:prefLabel
>     -- subfield $b : skos:notation ***
>     -- subfield $2 : skos:inScheme
>     For instance data that has no value in subfield $1 but has a value in subfield $b and a value in subfield $2 that is an RDA vocabulary , create an IRI based on the value in subfield $b, as follows:
>     -- If subfield $2 is a 'native' RDA source, for example 'rdaco' in field 336 : {RDA vocabulary IRI} + subfield $b, for example 'http://www.rdaregistry.info/termList/RDAContentType/' + '1001'.
>     -- If subfield $2 is an LoC clone of an RDA source, for example 'rdacontent' in field 336 : {LoC vocabulary IRI} + subfield $b, for example 'http://id.loc.gov/vocabulary/contentTypes/' + 'crd'. I think there are only three LoC cloned vocabularies, with the source codes 'rdacontent' (field 336). 'rdacarrier' (field 338), and 'rdamedia' (field 337), so these can be hard-wired in the transform. Use the appropriate object property with the created IRI and do not transform the other subfields.
>     For instance data that has no value in subfield $1 and no a value in subfield $2, use the appropriate RDA datatype property with subfield $a as the value and ignore the other subfields. If there is no subfield $a, use subfield $b as the literal value. If there is no subfield $a or $b, do not transform.
> 
> *** SKOS says that the notation is usually typed. We can do this by using/creating a local XML datatype based on the source given in subfield $2. The sources are coded using the LoC (Genre/Form Schemes)[https://id.loc.gov/vocabulary/genreFormSchemes.html]. We have discussed this use of the LoC source vocabularies in the context of 6XX fields.

**Comment by cspayne** [comment]
Cypress:

> @GordonDunsire @pennylenger
Both $a and $b are repeatable which means we can't know whether $a and $b will refer to the same concept. Will we have to mint separate concepts for each $a and $b?

Penny:

> Hi Cypress, I think we need to mint separate concepts for separate occurrences of subfield $a or subfield $b. And terms from different source vocabularies are recorded in separate occurrences of the field. 33X and 34X fields are also repeatable.

Gordon:

> @cspayne: The MARC 21 manual says to use repeats of subfield $a and $b only when they have the same source. If the sources are different, repeats of the field should be used. For fields where subfield $a is a controlled term and subfield $b is a code for the term, we should expect repeats to be ordered as $a, $b, $a, $b, etc. If this pattern is found, or the patterns $a, $a (with no $b) or $b, $b (with no $a), it is safe to assume that the repeats refer to different controlled terms and codes. If the pattern is $a, $a, $b or $a, $b, $b, then we should ignore the subfield $b and not add a skos:notation to the minted concept.

**Comment by cspayne** [comment]
Penny:

> Which options do we use for instance data that has no value in subfield $1 and $b, but has a value in subfield $a and a value in subfield $2 that is an RDA vocabulary?

Gordon:

> @pennylenger: if the field has a value in subfield $a and an RDA vocabulary source in subfield $2, we can safely mint an IRI for the RDA or cloned-RDA value depending on the source value. However, this will require a lookup of the term in subfield $a because the IRI patterns are based on the code/notation, not the term. If the term is not the preferred RDA label, the lookup will fail and an IRI cannot be created. I do not recommend, at this stage of the project, attempting to develop maps from 'old' terms to new. In any case, I don't think any of the controlled RDA terms have changed since they were added to the pre-3R RDA Toolkit, so any variations are not RDA, irrespective of what subfield $2 says. I don't know if LoC has changed any of the cloned terms since they were published.

Gordon:

> @pennylenger: Note also that if there are repeats of 33x fields, or of subfields within a 33x field, we are dealing with multiple unit or aggregate manifestations. A single unit manifestation can only be assigned one carrier type or one media type, and a single expression can only be assigned one content type if it is not an amalgamation expression such as a performed song. For 337 and 338, this means we have to process subfield $3 (materials specified). Good luck with that: the example in the MARC 21 manual is "liner notes", which is either a wrapper (considered to be an integral part of the carrier, such as the sleeve of a vinyl record) or an insert (considered to be a sub-unit of the carrier, such as an insert sheet of a vinyl record or a CD booklet). My recommendation is to not transform these fields if a subfield $3 is present.

**Reply by tmqdeborah** [reply]
@GordonDunsire, if a 336 has $atext $3liner notes, it should be safe to assume that the cataloger means an insert. If it has $3lyrics, then it could mean the wrapper, but we still want to know that lyrics are available as text. So I would recommend that we **do** transform the 336$a or $b when a $3 is also present.  And I would do the same for the 337 and 338, e.g.:

> 010    $a   2010483373
245 14 $a The MDS coordinator's field guide / $c Rena R. Shephard, MHA, RN, FACDONA, RAC-MT.
300    $a v, 211 pages : $b illustrations ; $c 29 cm + $e 1 CD-ROM (4 3/4 in.)
336    $3 book $2 rdacontent $a text
336    $3 CD $2 rdacontent $a text
337    $3 book $2 rdamedia $a unmediated
337    $3 CD $2 rdamedia $a computer
338    $3 book $2 rdacarrier $a volume
338    $3 CD $2 rdacarrier $a computer disc

> 010    $a    00517748 
> 245 00 $a Introspect / $c the Quiet Room.
> 300    $a 1 audio disc : $b digital ; $c 4 3/4 in.
> 336    $a performed music $b prm $2 rdacontent $3 audio disc
> 336    $a text $2 rdacontent $3 lyrics
> 337    $a audio $2 rdamedia $3 audio disc
> 337    $a unmediated $2 rdamedia $3 lyrics
> 338    $a audio disc $2 rdacarrier $3 audio disc
> 338    $a volume $2 rdacarrier $3 lyrics

It might also be useful to also record the 33X $a or $b AND the $3 in a Note on manifestation somehow, if we can come up with a way to make it understandable.

**Comment by pan-zhuo** [comment]
336 is trickier than 337 and 338. 337 and 338 are manifestation elements
where individual properties can add up to the whole, while 336 is at the
expression level where we don't know how many expressions there are, or
whether to mint an expression at all if it's an aggregate. Unless we mint
an aggregating work for everything and map 336 to content type of
representative expression. Not sure if we want to go that route.

**Comment by pennylenger** [comment]
I drafted a mapping based on previous discussions for 336. Please correct me if I've misunderstood anything or if there are any gaps in the logic.

**For pattern: $a$b$a$b... or all $a or all $b**
Look for the first $a, first $b, first $0 or first $1, then loop
If $1 is present or $0 contains "http":
[Expression] has content type [value of $1], ignoring all other subfields.

If no $1, but $2 (does not contain "rda"):
[Expression] has content type skos:Concept(mintedIRI)
<skos:Concept> skos:prefLabel <$a or $b if no $a>
<skos:Concept> skos:notation <$b>^^datatypeIRI(minted by us?)
<skos:Concept> skos:skos:inScheme <$2>

If no $1, but $2 (contains "rda"):
If $2 is "rdacontent":
[Expression] has content type "https://id.loc.gov/vocabulary/contentTypes.html" + subfield $b
If $a is "rdaco":
[Expression] has content type "http://www.rdaregistry.info/termList/RDAContentType/" + subfield $b

If no $1, no $b, but $a and $2 (contains "rda"):
Mint an IRI for the RDA or cloned-RDA value depending on the source value. However, this will require a lookup of the term in subfield $a because the IRI patterns are based on the code/notation, not the term.
(I am not sure here, do we look up the uri directly and insert the uri as value or do we mint a IRI?)

If no $1, no $2:
[Manifestation] has content type ["value of $a or $b if no $a"]

**Pattern: Not $a$b$a$b and not  all $a nor all $b, for example $a$b$b, or $b$a$a, then we should ignore the subfield $b and not add a skos:notation to the minted concept.** (I am not sure about this part. Do we ignore $b completely?)
Look for the first $a, first $0 and first $1, then loop
If $1 is present or $0 contains "http":
[Expression] has content type [value of $1], ignoring all other subfields.

If no $1, but $2 (does not contain "rda"):
[Expression] has content type skos:Concept(mintedIRI)
<skos:Concept> skos:prefLabel <$a>
<skos:Concept> skos:skos:inScheme <$2>

If no $1, but $a and $2 (contains "rda"):
Mint an IRI for the RDA or cloned-RDA value depending on the source value. 

If no $1, no $2:
[Manifestation] has content type ["value of $a"]

And for $3 Materials specified, should we add note on expression with the value like this: 
has content type “performed music” or “prm” or “http://rdaregistry.info/termList/RDAContentType/1011” apples to manifestation's $3? Or is it a note on manifestation? Adding the mapped element and value to the note.

**Reply by cspayne** [reply]
Thank you for putting this together Penny! It makes sense to me based on our discussions, although I added a few clarifications. 

> **For pattern: $a$b$a$b... or all $a or all $b**
> If $1 is present or $0 contains "http":
> [Expression] has content type [value of $1], ignoring all other subfields.

I think this would be:
[Expression] has content type [value of $1 or $0 containing "http"]
Also, If there is both $1 and $0, select the $1 value. 

> If no $1, but $2 (contains "rda"):
> If $2 is "rdacontent":
> [Expression] has content type "https://id.loc.gov/vocabulary/contentTypes.html" + subfield $b
> If $a is "rdaco":
> [Expression] has content type "http://www.rdaregistry.info/termList/RDAContentType/" + subfield $b

If $2 is "rdacontent", [Expression] has content type "https://id.loc.gov/vocabulary/contentTypes/" + subfield $b

Gordon will need to confirm, but I believe that when $2 is "rdaco", there won't be a $b, just an $a.

> If no $1, no $b, but $a and $2 (contains "rda"):
> Mint an IRI for the RDA or cloned-RDA value depending on the source value. However, this will require a lookup of the term in subfield $a because the IRI patterns are based on the code/notation, not the term.
> (I am not sure here, do we look up the uri directly and insert the uri as value or do we mint a IRI?)

These can be looked up in the transform at "https://id.loc.gov/vocabulary/contentTypes.rdf" when $2 is "rdacontent" and "http://www.rdaregistry.info/xml/termList/RDAContentType.xml" when $2 is "rdaco" to retrieve the IRI.

**Comment by GordonDunsire** [comment]
See also the discussion on subfield $3 (https://github.com/uwlib-cams/MARC2RDA/discussions/353).

The essential issue is the identification of the expression or manifestation that is the subject of the output triple(s). If we know that subfield $3 pertains to a subunit of the manifestation in focus, then the information on materials specified can be assigned to the manifestation being described (the superunit), but I think subfield $3 also covers separately published accompanying manifestations or embodied expressions, and then it's tricky.

**Comment by cspayne** [comment]
Our decision based on discussion in M2R meeting 2024-07-10:

The 3XX fields that map to a manifestation property (34Xs, 337, 338), $3 can be mapped to a note that states something like "[value] applies to: [$3]"
